app/src/main/java/io/github/sumukhshiv/memberquiz/GameActivity.java

```diff
@@ -0,0 +1,347 @@
+
+        //Add the wrong choices in the reminaing places
+        for (int i = 0,j = 0; i < buttonArray.length && j < wrongButtonChoices.length; i++) {
+            if (buttonArray[i].getText() == "") {
+                buttonArray[i].setText(wrongButtonChoices[j]);
+                j++;
+            }
+        }
+
+        //Set the image of the person
+        image.setImageResource(id);
+
+
+        //DISPLAY COUNTDOWN TIMER
+        timer = new CountDownTimer(5500, 1000) {
+
+            public void onTick(long millisUntilFinished) {
+                textViewTimer.setText("seconds remaining: " + millisUntilFinished / 1000);
+            }
+
+            public void onFinish() {
+                textViewTimer.setText("done!");
+                if (textViewTimer.getText().equals("done!")) {
+                    finish();
+                    startActivity(getIntent());
+                }
+
+            }
+        };
+
+        timer.start();
+
+        //New variable for contact
+        final String contactName = displayImageName;
+
+        //click the image to create a contact
+        image.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+
+                timer.cancel();
+                if (timer2 != null) {
+                    timer2.cancel();
+                }
+
+                // Creates a new Intent to insert a contact
+                Intent intent = new Intent(ContactsContract.Intents.Insert.ACTION);
+                // Sets the MIME type to match the Contacts Provider
+                intent.setType(ContactsContract.RawContacts.CONTENT_TYPE);
+
+                // Inserts the name of the contact
+                intent.putExtra(ContactsContract.Intents.Insert.NAME, contactName);
+                startActivity(intent);
+                textViewTimer.setText("seconds remaining: VOID");
+            }
+        });
+
+
+        //listeners for each of the four buttons
+        buttonA.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                String buttonText = buttonA.getText().toString();
+                if (!contactName.equalsIgnoreCase(buttonText)) {
+                    Toast.makeText(getApplicationContext(), "Incorrect!", Toast.LENGTH_SHORT).show();
+                } else {
+                    int userScore = pref.getInt("score", 0); // getting last score from the SharedPrefs, if there is no previous score it sets it to 0
+                    userScore++;
+                    SharedPreferences.Editor editor = pref.edit();
+                    editor.putInt("score", userScore); // Storing the new user score
+                    editor.commit(); //saving the changes
+                }
+                timer.cancel();
+                finish();
+                startActivity(getIntent());
+            }
+        });
+
+        buttonB.setOnClickListener(new View.OnClickListener() {
```

Comment:
If you're repeating something 4 times, it's probably better off in a private method